src/app/services/theme/theme.service.ts

```diff
@@ -0,0 +1,213 @@
+import { DOCUMENT } from "@angular/common";
+import { Inject, Injectable } from "@angular/core";
+import * as Color from "color";
+
+/** List of palette options for website theming */
+export enum BawPalettes {
+  highlight = "highlight",
+  primary = "primary",
+  secondary = "secondary",
+  success = "success",
+  info = "info",
+  warning = "warning",
+  danger = "danger",
+  light = "light",
+  dark = "dark",
+}
+
+/** List of variants which exist for each palette */
+enum BawVariants {
```

Comment:
```suggestion
enum BawColorVariants {
```

I don't believe this should be a public type, in which name spacing and specificity are not required